### ISSUE TITLE
Compiling from source no longer possible

### ISSUE THREAD

**Issue by FredrIQ** [rednex/rgbds#156]
As of commit eed098ac8e255f09b3727fa13ee577b7e0efada9, if I attempt to compile from source, I get the following errors.

gcc 6.3.1:

    src/asm/asmy.y: warning: 4 shift/reduce conflicts [-Wconflicts-sr]
    src/asm/asmy.y: warning: 3 reduce/reduce conflicts [-Wconflicts-rr]
    src/link/lexer.c:398:0: error: "yywrap" redefined [-Werror]
     
     
    src/link/lexer.c:74:0: note: this is the location of the previous definition
     
     
    cc1: all warnings being treated as errors
    make: *** [Makefile:106: src/link/lexer.o] Error 1

clang 3.9.1:

    src/asm/asmy.y: warning: 4 shift/reduce conflicts [-Wconflicts-sr]
    src/asm/asmy.y: warning: 3 reduce/reduce conflicts [-Wconflicts-rr]
    src/link/lexer.c:398:9: error: 'yywrap' macro redefined [-Werror,-Wmacro-redefined]
    #define yywrap() (/*CONSTCOND*/1)
            ^
    src/link/lexer.c:74:13: note: previous definition is here
        #define yywrap yywrap
                ^
    1 error generated.
        make: *** [Makefile:106: src/link/lexer.o] Error 1

I am using archlinux
Versions of applicable stuff
make: GNU make 4.2.1
gcc: 6.3.1/clang: 3.9.1
yacc: GNU bison 3.0.4
flex: 2.6.3
pkg-config: 0.29.1
libpng: 1.6.28

**Comment by AntonioND** [comment]
Ok, that's weird, considering that I had that same problem and I solved it with `%option noyywrap`.

Could you try to compile with `make FLAGS=" --nounistd --noyywrap"` or just `make FLAGS="--noyywrap"`?

**Comment by FredrIQ** [comment]
That just makes make throw out this a lot:

    make: Wall: Command not found

It's clearly misunderstanding flags in some way, not sure how though.

**Comment by AntonioND** [comment]
This is what I have:
```
$ gcc --version
gcc (Debian 6.3.0-11) 6.3.0 20170321
$ flex --version
flex 2.6.1
$ yacc --version
bison (GNU Bison) 3.0.4
```

**Comment by AntonioND** [comment]
Sorry, I meant `LFLAGS`.

`make LFLAGS="--nounistd --noyywrap"`
`make LFLAGS="--noyywrap"`?

**Comment by FredrIQ** [comment]
Same effect.

**Comment by AntonioND** [comment]
Weird, that works for me. What versions of flex and yacc do you have?

**Comment by FredrIQ** [comment]
yacc: GNU bison 3.0.4
flex: 2.6.3

**Comment by AntonioND** [comment]
Could you try adding `int yywrap (void) { return 1; }` to line 95 in `src/link/parser.y`, and removing line 19 in `src/link/lexer.l`? The one with `%option noyywrap`.

**Comment by FredrIQ** [comment]
OK, not sure what was up with that Wall spam, something I did earlier inadvertily broke anything make did in general (presumably some stray variable breaking stuff) even outside rgbds. Using a fresh instance made those issues go away, so I attempted your suggestions again. They didn't really do anything to solve the original issue though.

However, your last suggestion (manually editing files as you suggested) made make work properly.
